aten/src/THNN/generic/TemporalUpSamplingLinear.c

```diff
@@ -5,143 +5,25 @@
-    }
-    c10::raw::intrusive_ptr::decref(input);
-    return;
-  }
-  const accreal rwidth = linear_upsampling_compute_scale<accreal>(inputWidth, outputWidth, align_corners);
-  for (int w2 = 0; w2 < outputWidth; ++w2) {
-    const accreal w1r = linear_upsampling_compute_source_index<accreal>(rwidth, w2, align_corners);
-    const int w1 = w1r;
-    const int w1p = (w1 < inputWidth - 1) ? 1 : 0;
-    const scalar_t w1lambda = w1r - w1;
-    const scalar_t w0lambda = (scalar_t)1. - w1lambda;
-    const scalar_t* pos1 = &idata[w1];
-    // index w2 is interpolated by idata[w1] and (itself or idata[w1 + 1])
-    scalar_t* pos2 = &odata[w2];
-    for (int c = 0; c < channels; ++c) {
-      pos2[0] = w0lambda * pos1[0] + w1lambda * pos1[w1p];
-      pos1 += inputWidth;
-      pos2 += outputWidth;
-    }
-  }
-  c10::raw::intrusive_ptr::decref(input);
+    bool align_corners) {
+  THError('This function is deprecated, please use it from ATen.');
 }
 
 void THNN_(TemporalUpSamplingLinear_updateGradInput)(
-    THNNState *state,
-    THTensor *gradOutput,
-    THTensor *gradInput,
+    THNNState* state,
+    THTensor* gradOutput,
+    THTensor* gradInput,
     int nbatch,
     int channels,
     int inputWidth,
     int outputWidth,
-    bool align_corners){
-
-  THNN_(TemporalUpSamplingLinear_shapeCheck)
-    (NULL, gradOutput,
-     nbatch, channels,
-     inputWidth,
-     outputWidth);
-
-  THTensor_(resize3d)(gradInput, nbatch, channels, inputWidth);
-  THTensor_(zero)(gradInput);
-  gradOutput = THTensor_(newContiguous)(gradOutput);
-  scalar_t *data1 = gradInput->data<scalar_t>();
-  scalar_t *data2 = gradOutput->data<scalar_t>();
-  channels = nbatch * channels;
-
-  // special case: same-size matching grids
-  if (inputWidth == outputWidth) {
-    for (int w2 = 0; w2 < outputWidth; ++w2) {
-      const int w1 = w2;
-      scalar_t* pos1 = &data1[w1];
-      const scalar_t* pos2 = &data2[w2];
-      for (int c = 0; c < channels; ++c) {
-        pos1[0] += pos2[0];
-        pos1 += inputWidth;
-        pos2 += outputWidth;
-      }
-    }
-    c10::raw::intrusive_ptr::decref(gradOutput);
-    return;
-  }
-  const accreal rwidth = linear_upsampling_compute_scale<accreal>(inputWidth, outputWidth, align_corners);
-  for (int w2 = 0; w2 < outputWidth; ++w2) {
-    const accreal w1r = linear_upsampling_compute_source_index<accreal>(rwidth, w2, align_corners);
-    const int w1 = w1r;
-    const int w1p = (w1 < inputWidth - 1) ? 1 : 0;
-    const scalar_t w1lambda = w1r - w1;
-    const scalar_t w0lambda = (scalar_t)1. - w1lambda;
-    scalar_t* pos1 = &data1[w1];
-    const scalar_t* pos2 = &data2[w2];
-    for (int c = 0; c < channels; ++c) {
-      pos1[0] += w0lambda * pos2[0];
-      pos1[w1p] += w1lambda * pos2[0];
-      pos1 += inputWidth;
-      pos2 += outputWidth;
-    }
-  }
-  c10::raw::intrusive_ptr::decref(gradOutput);
+    bool align_corners) {
+  THError('This function is deprecated, please use it from ATen.');
 }
 
-#endif
+#endif
```

Comment:
Umm. Can you configure your editor to stop deleting the final newline? Many default configured editors (e.g., vim) will automatically add the trailing newline if it is missing, and this change will cause needless whitespace conflicts

Reply:
sure! thanks!